# HG changeset patch
# Parent 558d97abaa792b089381d5d69c5b28035c009be0
diff --git a/procinterrupts.c b/procinterrupts.c
--- a/procinterrupts.c
+++ b/procinterrupts.c
@@ -180,7 +180,15 @@ void parse_proc_interrupts(void)
 			break;
 		}
 
-		info->last_irq_count = info->irq_count;		
+		/* IRQ removed and reinserted, need restart or this will
+		 * cause an overflow and IRQ won't be rebalanced again
+		 */
+		if (count < info->irq_count) {
+			need_rescan = 1;
+			break;
+		}
+
+		info->last_irq_count = info->irq_count;
 		info->irq_count = count;
 
 		/* is interrupt MSI based? */
